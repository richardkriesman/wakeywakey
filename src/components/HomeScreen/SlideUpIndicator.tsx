import React, { ReactNode } from "react";
import { StyleSheet, View } from "react-native";

import { Button, Icon } from "react-native-elements";

/**
 * Properties that must be passed to <SlideUpIndicator />
 * @author Shawn Lutch
 */
export interface SlideUpIndicatorProps {
    // onPress event required
    onPress : VoidFunction;
}

/**
 * Indicator at the bottom of the screen to slide up.
 * @author Shawn Lutch
 */
export class SlideUpIndicator extends React.Component<SlideUpIndicatorProps> {

    protected onSwitch : VoidFunction;

    public constructor(props : SlideUpIndicatorProps) {
        super(props);
        this.onSwitch = props.onPress;
    }

    public render(): ReactNode {
        return (
            <Button
                buttonStyle={styles.indicator}
                onPress={this.onSwitch}
                icon={
                    <Icon
                        name="up"
                        type="antdesign"
                        size={32}
                        color="white"
                    />
                }
            />
        );
    }
};

const styles = StyleSheet.create({
    indicator: {
        backgroundColor: 'black',
        width: '100%'
    }
});
