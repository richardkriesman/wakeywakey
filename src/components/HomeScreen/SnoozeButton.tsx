/**
 * @module components
 */

import React, { ReactNode } from "react";
import { StyleSheet } from "react-native";
import { Button } from "react-native-elements";
import { Colors } from "../../constants/Colors";

export interface SnoozeButtonProps {
    onPress: VoidFunction;
}

export class SnoozeButton extends React.Component<SnoozeButtonProps> {
    public constructor(props: SnoozeButtonProps) {
        super(props);
    }

    public render(): ReactNode {
        return (
            <Button
                buttonStyle={styles.button}
                title="Snooze"
                onPress={this.props.onPress}
            />
        );
    }
}

const styles = StyleSheet.create({
    button: {
        backgroundColor: Colors.black,
        padding: 15
    }
});
