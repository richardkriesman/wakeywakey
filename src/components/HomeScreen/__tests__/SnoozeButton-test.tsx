import * as React from "react";
import { Button } from "react-native-elements";
import renderer from "react-test-renderer";
import { SlideUpIndicator } from "../SlideUpIndicator";
import { SnoozeButton } from "../SnoozeButton";

describe("SnoozeButton", () => {

    it("renders correctly", () => {
        const tree = renderer.create(<SnoozeButton onPress={() => void 0}/>);
        expect(tree).toMatchSnapshot();
    });

    it("reacts to onPress", () => {
        const resultText: string = "Alarm snoozed";

        const onPress = jest.fn();
        onPress.mockReturnValue(resultText);

        const tree = renderer.create(<SlideUpIndicator onPress={onPress}/>);
        expect(tree.root.findByType(Button).props.onPress()).toBe(resultText);
    });

});
