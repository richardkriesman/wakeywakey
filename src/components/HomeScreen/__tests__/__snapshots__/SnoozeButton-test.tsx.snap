// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnoozeButton renders correctly 1`] = `
<View
  style={
    Object {
      "borderRadius": 3,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#000000",
          "borderColor": "#2089dc",
          "borderRadius": 3,
          "borderWidth": 0,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 18,
            "paddingBottom": 1,
            "paddingTop": 2,
            "textAlign": "center",
          }
        }
      >
        Snooze
      </Text>
    </View>
  </View>
</View>
`;
